nativelib/src/main/scala/scala/scalanative/runtime/NativeThread.scala

```diff
@@ -93,43 +96,50 @@ object NativeThread {
   object Registry {
     import scala.collection.mutable
 
-    private val _aliveThreads = mutable.Set(currentNativeThread)
+    private val _aliveThreads = new ConcurrentHashMap[Long, NativeThread]()
```

Comment:
This probably won't work correctly until we replace the Scala.js `ConcurrentHashMap` with a port of JSR-166 implementation :)